astropy/coordinates/tests/test_regression.py

```diff
@@ -360,10 +362,68 @@ def test_regression_5743():
 def test_regression_5889_5890():
     # ensure we can represent all Representations and transform to ND frames
     greenwich = EarthLocation(
-        *u.Quantity([3980608.90246817, -102.47522911,  4966861.27310067],
+        *u.Quantity([3980608.90246817, -102.47522911, 4966861.27310067],
         unit=u.m))
     times = Time("2017-03-20T12:00:00") + np.linspace(-2,2,3)*u.hour
     moon = get_moon(times, location=greenwich)
     targets = SkyCoord([350.7*u.deg, 260.7*u.deg], [18.4*u.deg, 22.4*u.deg])
     targs2d = targets[:, np.newaxis]
     targs2d.transform_to(moon)
+
+
+def test_regression_6236():
+    # sunpy changes its representation upon initialisation of a frame,
+    # including via `realize_frame`. Ensure this works.
+    class MyFrame(BaseCoordinateFrame):
+        default_representation = CartesianRepresentation
+        my_attr = QuantityFrameAttribute(default=0, unit=u.m)
+
+    class MySpecialFrame(MyFrame):
+        def __init__(self, *args, **kwargs):
+            _rep_kwarg = kwargs.get('representation', None)
+            super(MyFrame, self).__init__(*args, **kwargs)
+            if not _rep_kwarg:
+                self.representation = self.default_representation
+                self._data = self.data.represent_as(self.representation)
```

Comment:
So we don't actually call `represent_as` in the `__init__` we just set the `representation` attribute and set `_data` equal to the rep we make.